SpatialGDK/Source/SpatialGDK/Private/Utils/ComponentFactory.cpp

```diff
@@ -204,14 +204,25 @@ void ComponentFactory::AddProperty(Schema_Object* Object, Schema_FieldId FieldId
 	}
 	else if (UObjectPropertyBase* ObjectProperty = Cast<UObjectPropertyBase>(Property))
 	{
-		UObject* ObjectValue = ObjectProperty->GetObjectPropertyValue(Data);
-
-		if (ObjectProperty->PropertyFlags & CPF_AlwaysInterested)
+		if (Cast<USoftObjectProperty>(Property))
 		{
-			bInterestHasChanged = true;
+			FSoftObjectPtr* ObjectPtr = (FSoftObjectPtr*)Data;
+
+			AddObjectRefToSchema(Object, FieldId, FUnrealObjectRef::FromSoftObjectPtr(*ObjectPtr));
 		}
+		else
+		{
+			UObject* ObjectValue = ObjectProperty->GetObjectPropertyValue(Data);
 
-		AddObjectRefToSchema(Object, FieldId, FUnrealObjectRef::FromObjectPtr(ObjectValue, PackageMap));
+			if (ObjectProperty->PropertyFlags & CPF_AlwaysInterested)
+			{
+				bInterestHasChanged = true;
+			}
+			if (ObjectValue)
+			{
+				AddObjectRefToSchema(Object, FieldId, FUnrealObjectRef::FromObjectPtr(ObjectValue, PackageMap));
```

Comment:
Should we print an Error here?

Reply:
Uh-Oh, i should send null pointers, wonder why I set it in a if(ObjectValue)

Reply:
:)